frontend/src/components/Dashboard/Opening/RecentOpeningsDataTable/index.tsx

```diff
@@ -0,0 +1,331 @@
+import React, { useContext, useEffect, useState } from "react";
+import {
+  TableToolbar,
+  TableToolbarAction,
+  TableToolbarContent,
+  TableToolbarMenu,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableHeader,
+  TableRow,
+  Button,
+  Pagination,
+  OverflowMenu,
+  OverflowMenuItem,
+  Popover,
+  PopoverContent,
+  Checkbox,
+  CheckboxGroup,
+  Modal,
+  ActionableNotification
+} from "@carbon/react";
+import * as Icons from "@carbon/icons-react";
+import StatusTag from "../../../StatusTag";
+import "./styles.scss";
+import EmptySection from "../../../EmptySection";
+import PaginationContext from "../../../../contexts/PaginationContext";
+import { OpeningsSearch } from "../../../../types/OpeningsSearch";
+import { ITableHeader } from "../../../../types/TableHeader";
+import { MenuItem } from "@carbon/react";
+import {
+  convertToCSV,
+  downloadCSV,
+  downloadPDF,
+  downloadXLSX,
+} from "../../../../utils/fileConversions";
+import { Tooltip } from "@carbon/react";
+import { useNavigate } from "react-router-dom";
+
+interface IRecentOpeningsDataTable {
+  rows: OpeningsSearch[];
+  headers: ITableHeader[];
+  defaultColumns: ITableHeader[];
+  handleCheckboxChange: Function;
+  setOpeningId: Function;
+  toggleSpatial: Function;
+  showSpatial: boolean;
+  totalItems: number;
+}
+
+const RecentOpeningsDataTable: React.FC<IRecentOpeningsDataTable> = ({
+  rows,
+  headers,
+  defaultColumns,
+  showSpatial,
+  totalItems,
+}) => {
+  const {
+    handlePageChange,
+    handleItemsPerPageChange,
+    itemsPerPage,
+    setInitialItemsPerPage,
+    currentPage,
+  } = useContext(PaginationContext);
+  const alignTwo = document?.dir === "rtl" ? "bottom-left" : "bottom-right";
+  const [openDownload, setOpenDownload] = useState(false);
+  const [selectedRows, setSelectedRows] = useState<string[]>([]); // State to store selected rows
+  const [toastText, setToastText] = useState<string | null>(null);
+  const [openingDetails, setOpeningDetails] = useState(false);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    setInitialItemsPerPage(itemsPerPage);
+  }, [rows, totalItems]);
+  
+  // Function to handle row selection changes
+  const handleRowSelectionChanged = (rowId: string) => {
+    setSelectedRows((prevSelectedRows) => {
+      if (prevSelectedRows.includes(rowId)) {
+        // If the row is already selected, remove it from the selected rows
+        return prevSelectedRows.filter((id) => id !== rowId);
+      } else {
+        // If the row is not selected, add it to the selected rows
+        return [...prevSelectedRows, rowId];
+      }
+    });
+  };
+
+  //Function to handle the favourite feature of the opening for a user
+  const handleFavouriteOpening = (rowId: string) => {
+    console.log(rowId + " has been added as a favourite for the user")
+    //make a call to the api for the favourite opening when ready
+    setToastText(`Following "OpeningID ${rowId}"`);
```

Comment:
Switch to the NotificationContext as described in [OpeningHistory](https://github.com/bcgov/nr-silva/blob/main/frontend/src/components/OpeningHistory/index.tsx#L25) to make it consistent with other notifications and to avoid code duplication

Reply:
done